File: BOJ_10845_큐/10845_허자용.cpp
#include <iostream>
#include <string>

using namespace std;

int len = 0;
int insert = 0;
int queue[10000];
int head = 0;

int push(int x) {
	queue[insert] = x;
	insert++;
	len++;
	return 0;
}

void pop() {
	if (!len) {
		cout << -1 << endl;
		return;
	}
	cout << queue[head] << endl;
	head++;
	len--;
	return;
}

void front() {
	if (!len) cout << -1 << endl;
	else cout << queue[head] << endl;
	return;
}

void back() {
	if (!len) cout << -1 << endl;
	else cout << queue[insert-1] << endl;
	return;
}


int main()
{	
	int T;
	cin >> T;
	string text;
	int a;

	for (int i = 0; i < T; i++) {
		cin >> text;
		if (text == "push") {
			cin >> a;
			push(a);
		}
		else if (text == "pop") pop();
		else if (text == "size") cout << len << endl;
		else if (text == "empty") {
			if (len) cout << 0 << endl;
			else cout << 1 << endl;
		}
		else if (text == "front") front();
		else if (text == "back") back();
	}
}

File: BOJ_10845_큐/10845_허자용.py
class Node:
    count = 0
    def __init__(self, value):
        self.value = value
        Node.count += 1
    def setprev(self, prev):
        self.prev = prev
    def setnext(self, next):
        self.next = next


T = int(input())

for i in range(T):
    text = input()
    if text[:4] == "push":
        pushvalue = int(text[5:])
        if not(Node.count):
            head = Node(pushvalue)
            tail = head
        else:
            newest = Node(pushvalue)
            newest.setprev(tail)
            tail.setnext(newest)
            tail = newest

    elif text == "pop":
        if not(Node.count):
            print(-1)
        else:
            print(head.value)
            Node.count -= 1
            if Node.count:
                head = head.next

    elif text == "size":
        print(Node.count)
        
    elif text == "empty":
        print(int(not(Node.count)))
        
    elif text == "front":
        if not(Node.count):
            print(-1)
        else:
            print(head.value)
            
    elif text == "back":
        if not(Node.count):
            print(-1)
        else:
            print(tail.value)
